src/couch_query_servers.erl

```diff
@@ -296,7 +296,8 @@ filter_docs(Req, Db, DDoc, FName, Docs) ->
     #httpd{} = HttpReq ->
         couch_httpd_external:json_req_obj(HttpReq, Db)
     end,
-    JsonDocs = [couch_doc:to_json_obj(Doc, [revs]) || Doc <- Docs],
+    Limit = config:get_integer("query_server_config", "filter_revs_limit", 20),
```

Comment:
You still think it only applicable for filter functions? How about this:
```
--- a/src/couch_query_servers.erl
+++ b/src/couch_query_servers.erl
@@ -282,7 +282,14 @@ validate_doc_update(DDoc, EditDoc, DiskDoc, Ctx, SecObj) ->
 
 json_doc(nil) -> null;
 json_doc(Doc) ->
-    couch_doc:to_json_obj(Doc, [revs]).
+    json_doc(Doc, json_doc_options()).
+json_doc(Doc, Options) ->
+    couch_doc:to_json_obj(Doc, Options).
+
+json_doc_options() ->
+    Limit = config:get_integer("query_server_config", "revs_limit", 20),
+    Options = [{revs, Limit}],
+    Options.
 
 filter_view(DDoc, VName, Docs) ->
     JsonDocs = [couch_doc:to_json_obj(Doc, [revs]) || Doc <- Docs],
@@ -296,7 +303,8 @@ filter_docs(Req, Db, DDoc, FName, Docs) ->
     #httpd{} = HttpReq ->
         couch_httpd_external:json_req_obj(HttpReq, Db)
     end,
-    JsonDocs = [couch_doc:to_json_obj(Doc, [revs]) || Doc <- Docs],
+    Options = json_doc_options(),
+    JsonDocs = [json_doc(Doc, Options) || Doc <- Docs],
     [true, Passes] = ddoc_prompt(DDoc, [<<"filters">>, FName],
         [JsonDocs, JsonReq]),
     {ok, Passes}.
```

Then this option will get shared for [shows](https://github.com/apache/couchdb-couch-mrview/blob/master/src/couch_mrview_show.erl#L78) and [updates](https://github.com/apache/couchdb-couch-mrview/blob/master/src/couch_mrview_show.erl#L123) functions.